semantic-analysis/src/Analysis/Syntax.hs

```diff
@@ -104,15 +130,19 @@ letrec n m = do
 
 -- Parsing
 
-parseFile :: (Has (Throw String) sig m, MonadIO m) => FilePath -> m (File Term)
+parseFile :: (Has (Throw String) sig m, MonadIO m) => FilePath -> m (Source.Source, File Term)
 parseFile path = do
   contents <- liftIO (B.readFile path)
+  let sourcePath = replaceExtensions path "py"
+  -- FIXME: this is comprised of several terrible assumptions.
+  sourceContents <- Source.fromUTF8 . B.toStrict <$> liftIO (B.readFile sourcePath)
```

Comment:
Is there a reason you’re using `Data.ByteString.Lazy`? It can be problematic.

Reply:
`aeson` 😞

In fact this one could probably be strict from the get-go, but demo code…